### PR TITLE
Hotfix/urllib to dev group

### PR DESCRIPTION
**Pull request by hjoaquim**
Change `pytest-recorder` to the dependencies' dev group instead, so that `urllib3` doesn't get "stuck" in a lower version.

### CODE REVIEW

**Comment by hjoaquim** [comment]
Platform running on Colab out of the box (with `urllib3` not conflicting):

image

**Review by piiq — CHANGES_REQUESTED** [review]
@hjoaquim thanks, can you look into the 2 comments i left in the diff, please?

File: openbb_platform/pyproject.toml [hunk 1]
```diff
@@ -83,6 +83,7 @@ all = [
 ]
 
 [tool.poetry.group.dev.dependencies]
+urllib3 = "<2.0.0"
```
[the remark this answers is not in the document]
Comment (hjoaquim) [reply openbb_platform/pyproject.toml:86]: @piiq you're correct, no need to have it here, pytest-recorder will bring a suitable version.

File: openbb_platform/dev_install.py [hunk 2]
```diff
@@ -78,7 +78,7 @@ def install_local(_extras: bool = False):
     """
     original_lock = LOCK.read_text()
     original_pyproject = PYPROJECT.read_text()
-    extras_args = ["-E", "all"] if _extras else []
+    extras_args = ["-E", "all", "--with", "dev"] if _extras else []
```
[the remark this answers is not in the document]
Comment (hjoaquim) [reply openbb_platform/dev_install.py]: @piiq you're correct once again on this one! Thanks for the details!

File: openbb_platform/platform/provider/pyproject.toml [hunk 3]
```diff
@@ -11,9 +11,11 @@ python = "^3.8"
 importlib_metadata = "^6.8.0"
 pydantic = "^2.4.2"
 requests = "^2.31.0"
-urllib3 = "<2.0.0"
```
Comment (piiq) [inline openbb_platform/platform/provider/pyproject.toml:14]: and here too?

File: openbb_platform/platform/provider/pyproject.toml [hunk 4]
```diff
@@ -11,9 +11,11 @@ python = "^3.8"
...
+urllib3 = "<3.0.0"
```
Comment (piiq) [inline openbb_platform/platform/provider/pyproject.toml:14]: i mean here

**Review by piiq — APPROVED** [review]

**Pull request merged**